src/source_map_utils.ts

```diff
@@ -30,7 +30,8 @@ export function extractInlineSourceMap(source: string): string {
     previousResult = result;
     result = inlineSourceMapRegex.exec(source);
   } while (result !== null);
-  const base64EncodedMap = previousResult![1];
+  if (!previousResult) return null;
```

Comment:
So the problem was that code calling into this has to be stateful - it has to first check if a source map is there, and then call this here. Which is easy to get wrong (qed), and then the failure mode is ugly on top.

I think you could improve the API by making it stateless, i.e. by merging the "has a comment" and "do something with it" calls. That makes calling code impossible to get wrong, at no loss of functionality (and actually a gain of performance, we save some parsing of the file).

The type system allows us to express the optionality, and you could rename the method to express that, e.g. `findInlineSourceMap`.